mypy/checker.py

```diff
@@ -5658,7 +5679,7 @@ def fail(self, msg: Union[str, ErrorMessage], context: Context, *,
              code: Optional[ErrorCode] = None) -> None:
         """Produce an error message."""
         if isinstance(msg, ErrorMessage):
-            self.msg.fail(msg.value, context, code=msg.code)
+            self.msg.fail(msg.value, context, code=code or msg.code)
```

Comment:
This branch works only when `isinstance(msg, ErrorMessage) is True`. Do we need to use `code` var here?

Looks like it does not fit this overload:

```python
@overload
    def check_subtype(self,
                      subtype: Type,
                      supertype: Type,
                      context: Context,
                      msg: ErrorMessage = ...,
                      subtype_label: Optional[str] = ...,
                      supertype_label: Optional[str] = ...,
                      *,
                      outer_context: Optional[Context] = ...) -> bool:
```

Reply:
Oh, that's actually from a different function, `TypeChecker.fail`, see the OP for more details.